src/execution/operator/persistent/physical_insert.cpp

```diff
@@ -425,9 +687,9 @@ idx_t PhysicalInsert::OnConflictHandling(TableCatalogEntry &table, ExecutionCont
 	// Check whether any conflicts arise, and if they all meet the conflict_target + condition
 	// If that's not the case - We throw the first error
 	idx_t updated_tuples = 0;
-	updated_tuples += HandleInsertConflicts<true>(table, context, lstate, data_table, *this);
+	updated_tuples += HandleInsertConflicts<true>(table, context, gstate, lstate, data_table, *this);
```

Comment:
Rather than modifying the way we do conflict handling here - I think it would be a lot cleaner to instead split up the input `DataChunk` prior to calling `HandleInsertConflicts` into two different chunks, that do not contain duplicate keys. For example, consider the following input chunk:

```csv
id	name
1	Pedro
1	Laurens
1	Mark
2	Thijs
3	Sam
```

For `ON CONFLICT DO NOTHING`, all we really need to do is remove all of the rows that have a conflict, and keep **only** the first row. We will end up with the following chunk:

```csv
id	name
1	Pedro
2	Thijs
3	Sam
```

For `ON CONFLICT DO UPDATE`, we further need to find the **last** row of every conflict, and call `HandleInsertConflicts` again using this second chunk. The second chunk will in the above example contain the following data:

```csv
id	name
1	Mark
```

Since we need to support this scenario anyway, this will allow us to support this case without needing to modify the on conflict handling code.

Reply:
That worked 👍 
Thanks for the suggestion, I was thinking about this the wrong way

This fixes both of the limitations I had previously mentioned in the PR description